project/Build.scala

```diff
@@ -549,6 +549,12 @@ object Build {
             IO.delete(trgDir)
           IO.createDirectory(trgDir)
           IO.unzip(scalaJSIRSourcesJar, trgDir)
+
+          // Remove f interpolator macro call to avoid its expansion while compiling the compiler and the implementation of the f macro
+          val utilsFile = trgDir / "org/scalajs/ir/Utils.scala"
+          val patchedSource = IO.read(utilsFile).replace("""f"\\u$c%04x"""", """"\\u%04x".format(c)""")
```

Comment:
This should be patched in the original source https://github.com/scala-js/scala-js/pull/3597